crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1227,70 +1227,202 @@ pub struct Segment2d {
 impl Primitive2d for Segment2d {}
 
 impl Segment2d {
-    /// Create a new `Segment2d` from its endpoints
+    /// Create a new `Segment2d` from its endpoints.
     #[inline(always)]
     pub const fn new(point1: Vec2, point2: Vec2) -> Self {
         Self {
             vertices: [point1, point2],
         }
     }
 
-    /// Create a new `Segment2d` from its endpoints and compute its geometric center
-    ///
-    /// # Panics
-    ///
-    /// Panics if `point1 == point2`
+    /// Create a new `Segment2d` from its endpoints and compute its geometric center.
     #[inline(always)]
     #[deprecated(since = "0.16.0", note = "Use the `new` constructor instead")]
     pub fn from_points(point1: Vec2, point2: Vec2) -> (Self, Vec2) {
         (Self::new(point1, point2), (point1 + point2) / 2.)
     }
 
-    /// Create a new `Segment2d` at the origin from a `direction` and `length`
+    /// Create a new `Segment2d` at the origin with the given direction and length.
+    ///
+    /// The endpoints will be at `-direction * length / 2.0` and `direction * length / 2.0`.
     #[inline(always)]
-    pub fn from_direction_and_length(direction: Dir2, length: f32) -> Segment2d {
-        let half_length = length / 2.;
-        Self::new(direction * -half_length, direction * half_length)
+    pub fn from_direction_and_length(direction: Dir2, length: f32) -> Self {
+        let endpoint = 0.5 * length * direction;
+        Self {
+            vertices: [-endpoint, endpoint],
+        }
     }
 
-    /// Get the position of the first point on the line segment
+    /// Create a new `Segment2d` at the origin from a vector representing
+    /// the direction and length of the line segment.
+    ///
+    /// The endpoints will be at `-scaled_direction / 2.0` and `scaled_direction / 2.0`.
+    #[inline(always)]
+    pub fn from_scaled_direction(scaled_direction: Dir2) -> Self {
```

Comment:
I'm not sure I understand what scaled direction means here.
The Dir2 is normalised anyway?

Reply:
Whoops, this was meant to be just a vector 🙈